pgmpy/factors/Factor.py

```diff
@@ -191,39 +190,22 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(), variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        marg_factor = []
```

Comment:
@palashahuja I don't think `maximize` is using `marg_factor` as any variable name. I think `marginalized_values` makes more sense.